src/main/java/io/mosip/digitalcard/service/impl/DigitalCardServiceImpl.java

```diff
@@ -241,17 +241,29 @@ private String getPassword(JSONObject jsonObject) throws Exception {
             if (obj instanceof JSONArray) {
                 // JSONArray node = JsonUtil.getJSONArray(demographicIdentity, value);
                 SimpleType[] jsonValues = Utility.mapJsonNodeToJavaObject(SimpleType.class, (JSONArray) obj);
-                uinCardPd = uinCardPd.concat(getParameter(jsonValues, templateLang).substring(0,4));
+                uinCardPd = uinCardPd.concat(getFormattedPasswordAttribute(getParameter(jsonValues, templateLang)).substring(0,4));
             } else if (object instanceof org.json.simple.JSONObject) {
                 org.json.simple.JSONObject json = (org.json.simple.JSONObject) object;
                 uinCardPd = uinCardPd.concat((String) json.get(VALUE));
             } else {
-                uinCardPd = uinCardPd.concat((String) object.toString().substring(0,4));
+                uinCardPd = uinCardPd.concat(getFormattedPasswordAttribute((String) object.toString()).substring(0,4));
             }
         }
         return uinCardPd;
     }
 
+    private String getFormattedPasswordAttribute(String password){
+        if(password.length()==3){
+            return password=password.concat(password.substring(0,1));
+        }else if(password.length()==2){
+            return password=password.repeat(2);
+        }else if(password.length()==1) {
+            return password=password.repeat(4);
+        }else {
+            return password;
+        }
```

Comment:
Rounak idea was to a take a configured value if the first name is shorter than 4, but here I see that we are repeating here. Was this idea discussed with Rounak ?

Reply:
yes this was discussed with Rounak and sasi